src/main/java/org/reaktivity/nukleus/http_cache/internal/proxy/cache/CacheEntry.java

```diff
@@ -235,20 +239,26 @@ private void sendResponseToClient(
 
     public void purge()
     {
-        this.state = CacheEntryState.PURGED;
-        if (clientCount == 0)
+        switch (this.state)
         {
-            cachedRequest.purge();
+            case PURGED:
+                break;
+            default:
+                this.state = CacheEntryState.PURGED;
+                if (clientCount == 0)
+                {
+                    cachedRequest.purge();
+                }
+                subscribers.stream().forEach(s ->
+                {
+                    MessageConsumer acceptReply = s.acceptReply();
+                    long acceptReplyStreamId = s.acceptReplyStreamId();
+                    long acceptCorrelationId = s.acceptCorrelationId();
+                    cache.writer.do503AndAbort(acceptReply, acceptReplyStreamId, acceptCorrelationId);
+                    s.purge();
+                });
+                subscribers.clear();
```

Comment:
Missing `break`

Reply:
this is fixed, see https://github.com/reaktivity/nukleus-http-cache.java/pull/44/commits/0e903c2ebddd73a6f218e0c137f4e07290d1ef8f